pkg/api/rest/rest.go

```diff
@@ -150,6 +151,16 @@ type Redirector interface {
 	ResourceLocation(ctx api.Context, id string) (remoteLocation *url.URL, transport http.RoundTripper, err error)
 }
 
+// Proxier supports proxying of connections
+type Proxier interface {
```

Comment:
How does this help me implement "GET /pod/foo/proxy"?  The sub resource "PodProxy" would implement Proxier *and* Get?  How would the Get method be implemented?  If the Get method is implemented, wouldn't it take priority over proxier?

Remember for proxying you eventually need the sub path, in which case you need to pass an alternate resource down to the call.

Here's an alternative:

```
type ObjectProxy interface {
    ProxyToObject(name string, w responsewriter, req request) (bool, error)
```

A Storage that implements this interface intends to proxy *ALL* requests to a given resource (<resourceType>/resourceName/*).  When used as a sub resource, the resource name is the parent resource name.  When not a sub resource, the path /<resourceType>/<resourceName> is exposed.

The RESTHandler will invoke the ProxyToObject method.  If an error is returned, request processing stops immediately (and the error is attempted to be written to the response stream).  If "true" is returned, the request was proxied and the handler should exit immediately.  If "false" is returned, the normal behavior for that resource path should be called.